src/services/powerSnapshotServices.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  findInCompletePowerSnapShots,
+  updatePowerSnapShots,
+} from '../repositories/powerSnapshotRepository';
+import { getRoundNumberByDate } from './cronJobs/syncUserPowers';
+import { getBlockByTime } from './blockByDateService';
+import { getTimestampInSeconds } from '../utils/utils';
+
+export const fillIncompletePowerSnapshots = async (): Promise<void> => {
+  const incompletePowerSnapshots = await findInCompletePowerSnapShots();
+  for (const powerSnapshot of incompletePowerSnapshots) {
+    const roundNumber = getRoundNumberByDate(
+      powerSnapshot.time,
+    ).previousGivbackRound;
+    const blockNumber = await getBlockByTime(
+      getTimestampInSeconds(powerSnapshot.time),
+    );
+    await updatePowerSnapShots({
```

Comment:
We can just fill the entity and call save() method on that.

Reply:
I didnt move the query to service file, but  as we talk I changed that in the repository funciton